### PR TITLE
Rename family in db

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR renames the actual database table names, foreign keys and constraints to use case instead of family. Pretty high risk PR since the schema changes are not contained in a transaction.

Let me know if you think it should be split into smaller PR:s isolating each schema change.

### Fixed
- Rename family to case in database schemas

### How to test

- [x] Apply upgrade and downgrade in stage
- [x] Ensure views work

```
(cg-env) ➜  cg git:(rename-family-in-db) ✗ alembic --config ../alembic-stage.ini upgrade head          
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 70e641d723b3 -> b105b426af99, Rename family in db
(cg-env) ➜  cg git:(rename-family-in-db) ✗ alembic --config ../alembic-stage.ini downgrade 70e641d723b3
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade b105b426af99 -> 70e641d723b3, Rename family in db
```

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def upgrade():
+    # Rename tables
+    op.rename_table("family", "case")
+    op.rename_table("family_sample", "case_sample")
+
+    # Drop foreign key constraints that reference the unique constraint to be dropped
+    op.drop_constraint("analysis_ibfk_1", "analysis", type_="foreignkey")
+    op.drop_constraint("case_sample_ibfk_1", "case_sample", type_="foreignkey")
```
Comment (ChrOertlin) [inline alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py:25]: should this be the other way around? Or does op.rename_table also alter the foreignkeys, that you then drop?

File: alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    # Rename tables back to original
+    op.rename_table("case", "family")
+    op.rename_table("case_sample", "family_sample")
+
+    # Recreate the original unique constraint
+    op.create_unique_constraint("_family_sample_uc", "family_sample", ["family_id", "sample_id"])
+
+    # Recreate foreign key constraints referencing the original unique constraint
+    op.create_foreign_key("analysis_ibfk_1", "analysis", "family", ["family_id"], ["id"])
+    op.create_foreign_key("case_sample_ibfk_1", "family_sample", "family", ["family_id"], ["id"])
```
Comment (ChrOertlin) [inline alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py:67]:
```suggestion
    op.create_foreign_key("family_sample_ibfk_1", "family_sample", "family", ["family_id"], ["id"])
```
right?

**Review by ChrOertlin — APPROVED** [review]
👍  Nice one question about the order in the alembic script. Looks solid otherwise

File: alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py (hunk 2 above)
Comment (seallard) [reply alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py:67]: Good catch!

File: alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py (hunk 1 above)
Comment (seallard) [reply alembic/versions/2023_11_08_b105b426af99_rename_family_in_db.py:25]: Yeah, I think it makes more sense to drop the constraints and then rename the table 👍

**Comment by seallard** [comment]
Deployed to vms and Hasta, stage and prod.

**Pull request merged**